packages/material-ui/src/utils/chainPropType.test.js

```diff
@@ -0,0 +1,45 @@
+import { assert } from 'chai';
+import PropTypes from 'prop-types';
+import { mock } from 'sinon';
+import chainPropType from './chainPropType';
+
+describe('chainPropType', () => {
+  const propName = 'children';
+  const props = { [propName]: 'some string' };
+  const componentName = 'ComponentName';
+  const location = 'prop';
+  let warning;
+
+  beforeEach(() => {
+    warning = mock(console).expects('error');
+  });
+
+  afterEach(() => {
+    warning.restore();
+  });
+
+  it('should have the right shape', () => {
+    assert.strictEqual(typeof chainPropType, 'function');
+  });
+
+  it('should not warn for supported types', () => {
+    PropTypes.checkPropTypes(
```

Comment:
Our tests are no longer idempotent with this method. The warning only displays once.

Reply:
Fixed it. See facebook/react#4302. Solution inspired by https://github.com/facebook/react/issues/7047#issuecomment-228614964